frontend/src/components/common/ImpactPerCause.tsx

```diff
@@ -0,0 +1,149 @@
+import React, { useState, useEffect } from "react";
+import { Box, Text, Flex, Button, Center, Circle } from "@chakra-ui/react";
+
+
+interface ItemData {
+  item_id: number;
+  item_name: string;
+  total_impact: number;
+}
+
+interface CauseData {
+  cause_id: number;
+  cause_name: string;
+  items: ItemData[]; 
+}
+
+const ImpactPerCause: React.FC = () => {
+  const [pageIndex, setPageIndex] = useState(0);
+  const [causes, setCauses] = useState<CauseData[]>([]);
+
+  // Buttons to move between different causes/pages
+  const handlePrevious = () => {
+    setPageIndex((prev) => (prev > 0 ? prev - 1 : prev));
+  };
+
+  const handleNext = () => {
+    setPageIndex((prev) => (prev < causes.length - 1 ? prev + 1 : prev));
+  };
+
+  // As of right now, impact is fetched from this link. In the future, this will be replaced by authContext.
+  // Change to localhost:5000 and modify the userId below based on the data entered in Prisma
+  async function fetchImpactData(userId: string) {
+    try {
+      const response = await fetch(`http://localhost:5001/impacts/${userId}`);
+      const data = await response.json();
+      console.log(data);
+      return data;
+    } catch (error) {
+      console.error("Error fetching impact data,", error);
+      return null;
+    }
+  }
+
+  useEffect(() => {
+    // Hardcoded in for testing purposes, change depending on what your user
+    const userId = "cly144mky0000bntg3dupxlx1";
+    fetchImpactData(userId).then((data) => {
+      setCauses(data);
+    });
+  }, []);
+
+  return (
+    <Box
+      maxW="40rem"
+      mx="auto"
+      p={5}
+      borderWidth="1px"
+      borderRadius="lg"
+      overflow="hidden"
+      borderColor="#E0DCDA"
+    >
+      {/* Header text displaying the cause name or loading message */}
+      <Text
+        fontSize="2xl"
+        mt={3}
+        mb={0}
+        textAlign="center"
+        fontWeight="bold"
+        color="#4D4D4D"
+      >
+        {causes.length > 0
+          ? `Total Impact For ${causes[pageIndex].cause_name}`
+          : "Loading Causes..."}
+      </Text>
+      {/* Content section to display items and navigation buttons */}
+      {causes.length > 0 && (
+        <Flex justifyContent="space-evenly" alignItems="center">
+          {/* Previous button */}
+          <Button onClick={handlePrevious} disabled={pageIndex === 0} background="transparent">
+            ❮
+          </Button>
+          <Box>
+            <Flex padding="5">
+              {causes[pageIndex].items.map((item) => {
+                const numDigits = item.total_impact.toString().length;
+                const width = `${65 + numDigits * 15}px`;
+                return (
+                  <Box
```

Comment:
Can you guys define these cards in a separate component?

It'll make the code a bit more modular, and also allow us to reuse them in other parts of the frontend.

Then, you can just import the card component into this file and render one for each item